driver-core/src/main/com/mongodb/client/model/changestream/UpdateDescription.java

```diff
@@ -56,15 +77,40 @@ public UpdateDescription(@Nullable @BsonProperty("removedFields") final List<Str
     }
 
     /**
-     * Returns the updatedFields
+     * Returns information about the updated fields excluding the fields reported via {@link #getTruncatedArrays()}.
      *
-     * @return the updatedFields
+     * @return {@code updatedFields}.
+     * @see #getTruncatedArrays()
      */
     @Nullable
     public BsonDocument getUpdatedFields() {
         return updatedFields;
     }
 
+    /**
+     * Returns information about the updated fields of the {@linkplain org.bson.BsonType#ARRAY array} type
+     * when the changes are reported as truncations.
+     *
+     * @return {@code truncatedArrays}.
+     * @see #getUpdatedFields()
+     */
+    @Nullable
+    public List<TruncatedArray> getTruncatedArrays() {
```

Comment:
Returning `null` from a method of a container type (`List` in this case) is an API design that not only violates the principle of least astonishment, but also forces users to write code that checks for both `null` and `isEmpty` instead of checking only `isEmpty`. Unfortunately, the test `ChangeStreamDocumentCodecSpecification.'should round trip ChangeStreamDocument successfully'` is written in such a way that prevents `UpdateDescription` from always returning an empty `List` instead of `null`. This in turn may be related to the behavior of BSON codec functionality.

Reply:
Perhaps change the test? Or always return a list from this method?

Reply:
> Perhaps change the test?

In order to change the test I would also need to implement the possibility to either not serialize empty lists in BSON or not deserialize empty lists from BSON. This comment simply explains a problem with the current `ChangeStreamDocument` and BSON serialization functionality, and states that the newly added method suffers from that same problem.

I think that if we want to improve the `ChangeStreamDocument` API (always return empty `List` values instead of returning `null values), it requires a separate task.

Reply:
I agree and think that always returning a list here is what is desirable.

Reply:
Done. Surprisingly, this did not require changing anything in the serialization mechanism.